### PR TITLE
Frost Select / Radio - Addressing several issues

### PR DESCRIPTION
**Pull request by srowhani**
# patch
# Changelog
- [x] There is a dead zone in the select between the text and the arrow. #183 
- [x] Please update CSS in Frost select #179
- [x] Frost Select throws error on enter without match #159
- [x] Using EventsProxy for events
- Frost Radio Group
  - Ember release 2.8 was erroring when destroying `parentView`
- Up arrow drawn in scss, one less element rendered
- Fix focus states by using attribute binding to `shouldOpen`
- If there are no display items, do not render `<div class='drop-down-container'>`

### CODE REVIEW

File: addon/components/frost-select.js [hunk 1]
```diff
@@ -384,11 +384,17 @@ export default Component.extend(PropTypeMixin, {
   // ==========================================================================
   // Events
   // ==========================================================================
-
+  focusIn (event) {
+    this.get('reduxStore').dispatch(openDropDown)
+    // If an onFocus event handler is defined, call it
+    if (this.attrs.onFocus) {
+      this.attrs.onFocus(event)
```
Comment (sandersky) [inline addon/components/frost-select.js]: Didn't we deprecate usage of `onFocus` on the other core components? It seems we ought to do the same here.

Comment (srowhani) [reply addon/components/frost-select.js]: Good catch

**Comment by sglanzer-fire** [comment]
Can you flag when this is ready for review again?

File: addon/components/frost-radio-button.js [hunk 2]
```diff
@@ -39,57 +40,82 @@ export default Component.extend(PropTypeMixin, {
...
     Ember.assert(
       `${this.toString()} must be initialized with a 'value' property`,
-      this.get('value')
+      get(this, 'value')
     )
   },
-
+  // fix for Ember 2.8 on application destroy
+  willDestroyElement () {
+    set(this, 'parentView', null)
```
Comment (srowhani) [inline addon/components/frost-radio-button.js:103]: `afterEach` hook from the `destroy-app` helper was throwing an error in 2.8. This addresses that.

Comment (sandersky) [reply addon/components/frost-radio-button.js:103]: Care to elaborate?

File: addon/components/frost-radio-button-input.js [hunk 3]
```diff
@@ -35,17 +36,15 @@ export default Component.extend(FrostEvents, PropTypeMixin, {
...
   // == Computed properties ====================================================
-  checked: computed('groupValue', 'value', function () {
-    return this.get('groupValue') === this.get('value')
-  }),
+
+  checked: alias('parentView.checked'),
 
   // == Functions ==============================================================
-  init () {
+  didInsertElement () {
```
Comment (juwara0) [inline addon/components/frost-radio-button-input.js:43]: Why the change from init()?

File: addon/components/frost-radio-button.js [hunk 4]
```diff
@@ -39,57 +40,82 @@ export default Component.extend(PropTypeMixin, {
...
-    target.id = this.get('groupId')
+    target.id = get(this, 'groupId')
     event.target = target
     return event
   },
 
   // == Events ===============================================================
 
-  init () {
+  didInsertElement () {
```
Comment (juwara0) [inline addon/components/frost-radio-button.js:91]: Same question as above, why the change from init()?

File: addon/components/frost-radio-button.js [hunk 5]
```diff
@@ -39,57 +40,82 @@ export default Component.extend(PropTypeMixin, {
...
-      if (this.get('disabled') || this.get('groupValue') === this.get('value')) {
+      if (get(this, 'disabled') || get(this, 'checked')) {
         return
       }
-      let change = this.get('onChange')
-      if (change && typeof change === 'function') {
-        change(this._createEvent(e, Ember.$(e.target).find('input')[0]))
+      const onChange = get(this, 'onChange')
+      if (typeOf(onChange) === 'function') {
+        onChange(this._createEvent(e, this.$(e.target).find('input')[0]))
```
Comment (juwara0) [inline addon/components/frost-radio-button.js:112]: use `Ember.$` and destructure at top. I saw a few other cases where `Ember.$` is used that can be destructured as well.

File: addon/components/frost-radio-button-input.js [hunk 6]
```diff
@@ -1,29 +1,30 @@
...
 const {
   Component,
-  computed
+  computed: {
+    alias
+  }
 } = Ember
 
-export default Component.extend(FrostEvents, PropTypeMixin, {
+export default Component.extend(PropTypeMixin, {
```
Comment (juwara0) [inline addon/components/frost-radio-button-input.js:11]: Why is `FrostEvents` mixin being removed?

**Comment by juwara0** [comment]
I would prefer to see this PR be focused on one area of changes instead of multiple changes over two different components. Keeping the PR focused makes the review easier to follow and if there are issues we can revert one set of changes without having to revert the entire PR with all of these changes.

**Comment by juwara0** [comment]
@srowhani I am in the process of writing the test suites for frost-core components. I have already started looking at frost-radio-\* and have some of the changes you are making in that changeset. Ping me later and we can chat about it.

File: addon/components/frost-radio-button-input.js (hunk 6 above)
Comment (sglanzer-fire) [reply addon/components/frost-radio-button-input.js:11]: I believe this is an internal construct (radio buttons need to be part of a group), so my assumption is that we would have the events/proxy on the group interface

File: addon/components/frost-radio-button.js (hunk 5 above)
Comment (sglanzer-fire) [reply addon/components/frost-radio-button.js:112]: There is a difference between Ember.$ (global scope) and this.$ (component/local scope) in case that matters

**Comment by sglanzer-fire** [comment]
Sorry to cause additional work @srowhani, but I agree with @juwara0 that it would be easier to merge this if the select/radio were separate PRs.  Would you mind splitting this?
